common/errors.go

```diff
@@ -46,29 +56,40 @@ func (e *Errors) GetErrors() []Error {
 	return e.errors[:]
 }
 
-// Append takes an Errors object as input creates a new Errors object with the current and input
-// errors.
+// Append creates a new Errors object with the current and input errors.
 func (e *Errors) Append(errs []Error) *Errors {
 	return &Errors{
-		errors: append(e.errors, errs...),
-		source: e.source,
+		errors:            append(e.errors, errs...),
+		source:            e.source,
+		numErrors:         e.numErrors + len(errs),
+		maxErrorsToReport: e.maxErrorsToReport,
 	}
 }
 
 // ToDisplayString returns the error set to a newline delimited string.
 func (e *Errors) ToDisplayString() string {
-	var result = ""
+	errorsTruncated := e.numErrors - e.maxErrorsToReport
+	errorsInString := int64(math.Min(float64(e.maxErrorsToReport), float64(e.numErrors)))
```

Comment:
Well this is gross. Maybe just hand write it instead of casting to float and back?

Reply:
Hah, yes it is. I've fixed it by just using if-else blocks instead.